### PR TITLE
Feature/tsp visibility sc 2173

### PR DESCRIPTION
**Pull request by tofixx**
# Checkliste

- Bis die komplette Checkliste abgearbeitet ist muss das PR-Label WIP gesetzt sein

## Allgemein
- [x] Links zu verwandten PRs anderer Repositories
  - https://github.com/schul-cloud/schulcloud-server/pull/887
  - Im Ticket (oder PR, wenn kein Ticket): Beschreibung und Begründung der Änderung/Neuerungen
- [x] Link zum Ticket https://ticketsystem.schul-cloud.org/browse/SC-2173

### CODE REVIEW

File: config/default.schema.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		"TEACHER_VISIBILITY_FOR_EXTERNAL_TEAM_INVITATION": {
+			"type": "string",
+			"default": "opt-out",
+			"enum": [
+				"disabled",
+				"opt-in",
+				"opt-out",
+				"enabled"
+			],
+			"description": "defines wheter external team invitation shows teachers from different schools or not. if enabled system wide there are options general enabled or opt-in/-out by user required."
```
Comment (Parnswir) [inline config/default.schema.json]: See my suggestion in the server PR

File: controllers/account.js [hunk 2]
```diff
@@ -16,24 +16,28 @@ router.post('/', (req, res) => {
...
 		passwordNew,
 	} = req.body;
-	const discoverable = !!req.body.discoverable;
+	let discoverable = null;
+	if (req.body.discoverable === 'true') {
+		discoverable = true;
+	}
+	if (req.body.discoverable === 'false') {
+		discoverable = false;
+	}
```
Comment (adrianjost) [inline controllers/account.js]: shouldn't `discoverable` default to `false`? `null´ is handled like `false` in our code so it is unnecessary to have the second if.

File: app.js [hunk 3]
```diff
@@ -13,13 +13,18 @@ const handlebars = require('handlebars');
 const layouts = require('handlebars-layouts');
 const handlebarsWax = require('handlebars-wax');
 const Sentry = require('@sentry/node');
+const { Configuration } = require('@schul-cloud/commons');
```
Comment (adrianjost) [inline app.js]: `Configuration` is really generic and difficult to understand. I have no clue what this package is doing.
After reading the npm package description I understand what it should do, but I still think the name should be changed here.
There is also no documentation at all, how to use and configure that package.

File: controllers/account.js (hunk 2 above)
Comment (tofixx) [reply controllers/account.js]: nope, we have a nullable boolean here so it has three states, not only two.

**Pull request merged**